### ISSUE TITLE
Empty collection error

### ISSUE THREAD

**Issue by yazfield** [yajra/laravel-datatables#149]
I'm getting an error when i try to get data from an empty table, the error is:

>ErrorException in CollectionEngine.php line 46:
>array_keys() expects parameter 1 to be array, null given

My controller contains two simple functions
```php
// get the view
public function getItems() {
  return view('someview');
}

// get data for datatables
public function getItemsdata() {
  $data = DataModel::all();
  return Datatables::of($data)->make(true);
}
```

When $data is not empty, everything works just fine but if it is empty, I get the error! how can i fix this?

**Comment by yazfield** [comment]
@yajra any workaround?

**Comment by yajra** [comment]
@yazfield, what version are you using? I just tested the same code on my demo app with empty records and it seems to work fine.

```php
    public function getBasicObject()
    {
        return view('datatables.collection.basic-object');
    }

    public function getBasicObjectData()
    {
        $users = User::all();

        return Datatables::of($users)->make(true);
    }
```

**Comment by yazfield** [comment]
i was using Laravel 5.0 & datatables 5.1.1, switched to datatables 5.0 and now i'm getting another error when the collection is empty:

>ErrorException in BaseEngine.php line 244:
>Undefined offset: 0

**Comment by yazfield** [comment]
knowing that my js is something like:

```javascript
var table = $('#table').DataTable({
                processing: true,
                serverSide: true,
                ajax: '/itemsdata',
                columns: [
                     {data: 'level', name: 'level'}
                ]
       });
```

**Comment by yajra** [comment]
Latest version is Datatables 5.11.1. Can you try using the latest version please?

**Comment by yazfield** [comment]
@yajra it seems the problem is in the JS, i'm adding a checkbox column for selection this way:
```javascript
var table = $('#table').DataTable({
                processing: true,
                serverSide: true,
                ajax: '/itemsdata',
                columns: [
                     {data: null, render: function(data,type,full){
                           return '<input type="checkbox" />';
                     }}
                     {data: 'column', name: 'column'}
                ]
       });
```
this works just fine when the Collection is not empty, but issues an error if it's empty, i've been on it for several days checking datatables, any ideas please?

**Comment by yajra** [comment]
I am not familiar with the render option but what I usually do when I add a checkbox is this:

```js
{
data: function(data) {
    return '<input type="checkbox" />';
  }
}
```

**Comment by yajra** [comment]
@yazfield, I released v5.11.8 that should fix empty collection issue. Kindly checkout and confirm. Thanks and sorry for the late resolution!
